src/solvers/z3/z3_conv.cpp

```diff
@@ -18,7 +18,11 @@ static void error_handler(Z3_context c, Z3_error_code e)
   oss << "Z3 error " << e << " encountered"
       << "\n";
   oss << Z3_get_error_msg(c, e);
-  assert(0 && oss.str().c_str());
+  default_message msg;
+  // use msg.status + assert instead of msg.error
+  // In GDB, the latter does not show a backtrack of the exact failure line in this function
```

Comment:
I believe you meant `abort` instead of `assert`. Why does msg.error not work?

Reply:
Done. 

> Why does msg.error not work?

Given that `error_handler` is a callback function we passed into Z3_set_error_handler and `msg.error` does not exit on failure, my conjecture is that if we don't exit in `error_handler` ESBMC will continue in some Z3 library routine. However, GDB may not have the Z3 debugging symbols to map to the corresponding function/line in the source file.

Reply:
Changed to msg.error + abort and removed the comments. 

Sorry about the confusing comments. Previously, I had always been mistaken about `msg.error` aborting the execution automatically. The comment actually means using msg.error **alone** would not make the program abort and hence not showing a clear backtrace of the exact line.